packages/wrangler/src/api/dev.ts

```diff
@@ -80,7 +80,7 @@ export async function unstable_dev(
 	const { testMode = true, disableExperimentalWarning = false } =
 		options?.experimental ?? {};
 	if (apiOptions) {
-		throw logger.error(
+		logger.error(
```

Comment:
per our discussion, I think this is the better option to prevent unnecessary breakage while still making the API change.